library/common/engine.h

```diff
@@ -39,10 +39,30 @@ class Engine {
    * Increment a counter with a given string of elements and by the given count.
    * @param elements, joined elements of the timeseries.
    * @param count, amount to add to the counter.
-   * @param envoy_status_t, the resulting status of the operation.
    */
   envoy_status_t recordCounter(const std::string& elements, uint64_t count);
 
+  /**
+   * Set a gauge of a given string of elements with the given value.
+   * @param elements, joined elements of the timeseries.
+   * @param value, value to set to the gauge.
+   */
+  envoy_status_t setGauge(const std::string& elements, uint64_t value);
```

Comment:
I had originally been thinking we could group all stats-recording functions by prefixing them with `record*`. These could be called `recordGuageSet`, `recordGuageAdd`, `recordGuageSub`. `recordCounter` could be change to `recordCounterInc` or stay the same. What do you think?

Reply:
Grouping all the stats-recording functions by the same prefix sounds a good idea to me. 

I'll update the gauge related methods to the suggested names in this pr. 

As for the renaming of `recordCounter`, I assume we want the renaming to go all the way to the platform apis. It will be a breaking change, and I will split that change into a separate pr.

How does this plan sound? @goaway